### ISSUE TITLE
Public API to hide the dockLayer Side panel

### ISSUE THREAD

**Issue by kolibril13** [napari/napari#4412]
As discussed in https://github.com/napari/napari/pull/3947, it would be amazing to hide the side panel in napari programmatically.

This is already possible by accessing the private API like this:
```
viewer.window._qt_viewer.dockLayerControls.toggleViewAction().trigger()
viewer.window._qt_viewer.dockLayerList.toggleViewAction().trigger()
```
It would more convenient though to create a new public API, e.g. as @jni suggested:
```
viewer.controls.visible = False 
viewer.layers.visible = False
```

Why this would be nice to have:
One could then use `viewer.window.resize(600, 600)` together with `viewer.screenshot()` and the aspect ratio would be kept.
To keep the aspect ratio, one could also use right now the `size= ` parameter in `screenshot`, but that comes with disadvantage that it is not wysiwyg.

**Comment by cnstt** [comment]
I have exactly the same request (this command is needed for plugins) and I currently get this warning each time I use the private API:

`FutureWarning: Public access to Window.qt_viewer is deprecated and will be removed in
v0.5.0. It is considered an "implementation detail" of the napari
application, not part of the napari viewer model. If your use case
requires access to qt_viewer, please open an issue to discuss.!
`

**Comment by cnstt** [comment]
As in https://github.com/napari/napari/issues/4683 I would like to know if this is something you would want to see implemented as a public API @alisterburt @Czaki @tlambert03 or others :)
Because I could try to work on it, but I would rather want to know if this is wanted before starting the work ;)

**Comment by tlambert03** [comment]
I agree we probably need some public API to generally achieve the hiding and showing of widgets.  But it would be nice to do so in a single method such that we don't need to add a new one everytime we want to show hide another thing.  `viewer.controls.visible = False` and `viewer.layers.visible = False` are simple enough for now, but what about `viewer.console` ... and yes we have two commonly used internal controls for now, but will we always?  and do they each have to come with a new public method? 

Perhaps we could consider a more future proof `viewer.window.hide/show_widget(some_key)`?  (note: I don't _love_ that particular proposal, especially if `some_key` is a string ... but I also don't know that I love having a new `viewer.window.thing` for every new widget we want to make available

**Comment by Modjular** [comment]
Is this api decision still under consideration as `0.5.0` approaches? I've been using [this method from the forums](https://forum.image.sc/t/is-it-possible-to-open-the-napari-viewer-with-the-layer-controls-and-layer-list-gui-elements-hidden/47755/2?u=anthony_reksoatmodjo)

```python
import napari

with napari.gui_qt():
    viewer = napari.Viewer()
    viewer.window.qt_viewer.dockLayerList.setVisible(False)
    viewer.window.qt_viewer.dockLayerControls.setVisible(False)
```

but it throws the `FutureWarning` mentioned above.

```
Public access to Window.qt_viewer is deprecated and will be removed in
v0.5.0. It is considered an "implementation detail" of the napari
application, not part of the napari viewer model. If your use case
requires access to qt_viewer, please open an issue to discuss.
```

**Comment by Czaki** [comment]
There is still a lack of implementation of the necessary API that will allow to removal `viewer.qt_viewer`. So we do not know at this moment what will happen. most probably not at 0.5.0. maybe 0.5.1 or 0.6.0 

But If you do not want LayerList and LayerControls then you may directly use `QtViewer` instead of `Viewer` and embed it in your own interface.
